cache/redis_cache.go

```diff
@@ -310,3 +329,10 @@ type RedisCacheError struct {
 func (e *RedisCacheError) Error() string {
 	return fmt.Sprintf("error while reading cached result in redis, only %d bytes of %d were fetched", e.readPayloadSize, e.expectedPayloadSize)
 }
+
+type RedisCacheCorruptionError struct {
+}
+
+func (e *RedisCacheCorruptionError) Error() string {
+	return "chproxy can't decode the cached result from redis, it seems to have been corrupted"
```

Comment:
May be it'd be worth to add an info about the key in order to ease analysing the payload